cmd/create/cluster/cmd.go

```diff
@@ -871,27 +899,39 @@ func run(cmd *cobra.Command, _ []string) {
 		args.ec2MetadataHttpTokens = string(v1.Ec2MetadataHttpTokensOptional)
 	}
 
+	// Errors when users elects for cluster admin via flags and elects for hosted control plane via interactive prompt"
+	if isHostedCP && (createAdminUser || createAdminPassword != "") {
+		r.Reporter.Errorf("Setting Cluster Admin is only supported in classic ROSA clusters")
+		os.Exit(1)
+	}
+
+	// isClusterAdmin is aflag indicating if user wishes to create cluster admin
 	isClusterAdmin := false
 	if !isHostedCP {
-		if clusterAdminUser != "" {
-			r.Reporter.Warnf("cluster admin user is %s", clusterAdminUser)
-			err = idp.UsernameValidator(clusterAdminUser)
-			if err != nil {
-				r.Reporter.Errorf("%s", err)
-				os.Exit(1)
-			}
+		if createAdminUser {
 			isClusterAdmin = true
+			// user supplies create-admin-user flag without create-admin-password will generate random password
+			if createAdminPassword == "" {
+				r.Reporter.Debugf("Generating random password")
+				createAdminPassword, err = admin.GenerateRandomPassword(23)
```

Comment:
NP
Can we please use a const here?

Reply:
I will go ahead and add it to the create/admin file as the GenerateRandomPassword method also originates from there.